### ISSUE TITLE
Create sample chaincode in Go and test it in CI

### ISSUE THREAD

**Issue by dzikowski** [hyperledger-labs/fablo#99]
Any sample could be used like this one for example

https://github.com/hyperledger/fabric-samples/tree/main/asset-transfer-basic/chaincode-go

**Comment by joseucarvajal** [comment]
+1

**Comment by debayangg** [comment]
@dzikowski I would like to work on this issue. I went through all the samples and saw that this [config](https://github.com/hyperledger-labs/fablo/blob/main/samples/fablo-config-hlf2-2orgs-2chaincodes-raft.yaml) uses two chaincodes one in nodejs and another in java which is later tested [here](https://github.com/hyperledger-labs/fablo/blob/main/e2e-network/docker/test-02-raft-2orgs.sh) . Should I follow these and create a sample chaincode in Go and test it and later add it to GH actions.

**Comment by dzikowski** [comment]
yes, it would be perfect this way! You can even remove the node.js chaincode from that config and replace it with the one in Go. Would be leaner, and we test Node.js chaincode also in other tests

**Comment by kalio007** [comment]
@dzikowski if this is cold, i would love to handle this

**Comment by debayangg** [comment]
@kalio007 was working on this, was able to create a simple golang chaincode with a test but I have been facing problems with the test-02-raft-2orgs.sh integrating go chaincode with java

**Comment by kalio007** [comment]
Thanks for the update, do you mind sharing your current changes to the PR so someone can take a look and help out?

**Comment by debayangg** [comment]
The basic tests itself were failing so I didnt create a PR and was investigating but okay then I can create a draft PR for now,

**Comment by debayangg** [comment]
@dzikowski since I am facing some issues and have to investigate them, can I just create a simple chaincode and yaml setup for golang exclusively and test it with a custom shell script that conforms to the style in e2e-network/docker. Also can you provide some context regarding the e2e tests and snapshots.

**Comment by dzikowski** [comment]
Is the relevant e2e-network test script working? If yes, then it might be just a matter of updating snapshots with `npm run test:e2e-update` (please check also our contribution guide)

**Comment by debayangg** [comment]
Okay I will update the e2e tests and check again

**Comment by debayangg** [comment]
I tried investigating further and it seems to be mainly due to mismatch between fabric and the go chaincode versions, as documented in the following issues:
1. https://github.com/hyperledger/fabric/issues/4951
2. https://github.com/hyperledger/fabric/issues/3039

I have tried go mod vendor but it didn't fix this issue.

**Comment by joaormatos** [comment]
> Any sample could be used like this one for example
> 
> https://github.com/hyperledger-bevel/bevel-operator-fabric/tree/v1.11.0-beta7/examples/client/go

Note: the URL above (added to OP by @umegbewe in an edit 2 weeks ago) links to a client made using the old Fabric SDK, not a chaincode sample.

Examples of Go chaincode can be found in the fabric-samples repository.
For example: https://github.com/hyperledger/fabric-samples/tree/main/asset-transfer-basic/chaincode-go

**Comment by dzikowski** [comment]
already resolved
